nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+const knex = require("../database");
+const express = require("express");
+const router = express.Router();
+//get method
+router.get("/meals", async (req, res) => {
+          try {
+                    const allMeals = await knex.from("meal").select("*");
+                    if (allMeals.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists any more"
+                              );
+                    }
+                    res.send(allMeals);
+          } catch (err) {
+                    throw err;
+          }
+});
+//get method using id
+router.get("/meals/:id", async (req, res) => {
+          try {
+                    const mealId = req.params.id;
+                    const mealWithId = await knex
+                              .from("meal")
+                              .select("*")
+                              .where({ meal_id: mealId });
+                    if (mealWithId.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists"
+                              );
+                    }
+                    res.send(mealWithId);
+          } catch (err) {
+                    throw err;
+          }
+});
+//post method
+router.post("/meals", async (req, res) => {
+          const query = {
+                    title: req.query.title,
+                    description: req.query.description,
+                    location: req.query.location,
+                    price: req.query.price,
+          };
+          if (query) {
+                    const qResponse = await knex("meal").insert(query);
+                    if (qResponse) {
+                              res.status(404).send("data added");
```

Comment:
is 404 the right status code to send when creating a meal is successful?

Reply:
And in regards to the body a common pattern in REST when you create a new resource or entity like a meal is to return the object itself including its new id. That way the client that is using this new meal can use that id in the future if it needs to delete it, update it or obtain updates done by other clients.